optaplanner-core/src/main/java/org/optaplanner/core/impl/score/stream/bavet/common/BavetNodeBuildPolicy.java

```diff
@@ -77,4 +77,22 @@ public int getNodeOrderMaximum() {
         return joinConstraintStreamToJoinBridgeNodeMap;
     }
 
+    public List<BavetNode> getCreatedNodes() {
+        // Make a sequential list of unique nodes.
+        SortedMap<Integer, BavetNode> nodeIndexToNodeMap = Stream.concat(sharableNodeMap.keySet().stream(),
+                constraintIdToScoringNodeMap.values().stream())
+                .distinct()
```

Comment:
Why is the distinct needed on the nodes? I'd argue if they are not distinct, then node sharing is broken and we want it to fail fast.